servlet/src/jettyTest/java/io/grpc/servlet/JettyInteropTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.servlet;
+
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.ServerBuilder;
+import io.grpc.netty.InternalNettyChannelBuilder;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.testing.integration.AbstractInteropTest;
+import org.eclipse.jetty.http2.parser.RateControl;
+import org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class JettyInteropTest extends AbstractInteropTest {
+
+  private static final String HOST = "localhost";
+  private static final String MYAPP = "/grpc.testing.TestService";
+  private int port;
+  private Server server;
+
+  @After
+  @Override
+  public void tearDown() {
+    super.tearDown();
+    try {
+      server.stop();
+    } catch (Exception e) {
+      throw new AssertionError(e);
+    }
+  }
+
+  @Override
+  protected ServerBuilder<?> getServerBuilder() {
+    return new ServletServerBuilder().maxInboundMessageSize(AbstractInteropTest.MAX_MESSAGE_SIZE);
+  }
+
+  @Override
+  protected void startServer(ServerBuilder<?> builer) {
+    GrpcServlet grpcServlet =
+        new GrpcServlet(((ServletServerBuilder) builer).buildServletAdapter());
+    server = new Server(0);
+    ServerConnector sc = (ServerConnector)server.getConnectors()[0];
+    HTTP2CServerConnectionFactory factory =
+            new HTTP2CServerConnectionFactory(new HttpConfiguration());
+    factory.setRateControlFactory(new RateControl.Factory() {});
```

Comment:
(I'm not familiar with Jetty) What's the reason to add rate control?

Reply:
As with tomcat, jetty defaults to some degree of protection here, and also needs to be turned down. Jetty 9 has no errors, but in Jetty 10, the veryLargeResponse test fails with the server canceling the stream in the JettyInteropTest.